### PR TITLE
Add support for field data

### PR DESCRIPTION
**Pull request by tverho**
andresegido contributed a pull request to add support for field data (https://github.com/JuliaVTK/ReadVTK.jl/pull/42). However, due to the lack of a test, it was not merged. Here's a new pull request with a test included. Additionally, the `get_field_data` function is exported. I hope this can be now merged.

Closes #42

### CODE REVIEW

File: src/ReadVTK.jl [hunk 1]
```diff
@@ -433,6 +439,29 @@ function get_data_section(vtk_file::VTKFile, section)
   return VTKData(names, data_arrays, vtk_file)
 end
 
+"""
+    get_field_data(vtk_file::VTKFile)
+
+Retrieve a lightweight `VTKData` object with the field data of the given VTK file.
+
+See also: [`VTKData`](@ref), [`get_point_data`](@ref), [`get_cell_data`](@ref)
```
Comment (ranocha) [inline src/ReadVTK.jl:447]: Can you please add similar cross-references to other functions, e.g., reference `get_field_data` from the docstring of `get_point_data` etc.?

File: src/ReadVTK.jl [hunk 2]
```diff
@@ -433,6 +439,29 @@ function get_data_section(vtk_file::VTKFile, section)
...
+"""
+function get_field_data(vtk_file::VTKFile)
+  names = String[]
+  data_arrays = XMLElement[]
+
+  # Iterate over XML elemens in the section
+  for xml_element in child_elements(field_data(vtk_file))
+    # We do not know how to handle anything other than `DataArray`s
+    @assert LightXML.name(xml_element) == "DataArray"
```
Comment (ranocha) [inline src/ReadVTK.jl:456]:
```suggestion
    LightXML.name(xml_element) != "DataArray" && continue
```

See https://github.com/JuliaVTK/ReadVTK.jl/pull/65

**Review by ranocha — CHANGES_REQUESTED** [review]
Thanks! I have two minor suggestions.

File: src/ReadVTK.jl (hunk 1 above)
Comment (tverho) [reply src/ReadVTK.jl:447]: Done

**Review by ranocha — COMMENTED** [review]
Thanks a lot! Could you please add field data to

- https://github.com/JuliaVTK/ReadVTK.jl?tab=readme-ov-file#what-works
- https://github.com/JuliaVTK/ReadVTK.jl/blob/main/docs/src/index.md#what-works

and run the formatter locally to fix the CI run

https://github.com/JuliaVTK/ReadVTK.jl/actions/runs/11518148027/job/32122006751?pr=66

so that all required CI checks pass?

**Review by ranocha — APPROVED** [review]
Thanks a lot!

File: test/fielddata.jl [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Create some cells
+points = rand(3, 5)
+cells = [
+  MeshCell(VTKCellTypes.VTK_TRIANGLE, [1, 4, 2]),
+  MeshCell(VTKCellTypes.VTK_QUAD, [2, 4, 3, 5])
+]
```
Comment (ranocha) [inline test/fielddata.jl:6]:
```suggestion
cells = [
  MeshCell(VTKCellTypes.VTK_TRIANGLE, [1, 4, 2]),
  MeshCell(VTKCellTypes.VTK_QUAD, [2, 4, 3, 5]),
]
```

**Pull request merged**